lib/diaspora_federation/entities/relayable.rb

```diff
@@ -4,10 +4,56 @@ module Relayable
       def self.included(model)
         model.class_eval do
           property :parent_guid
-          property :parent_author_signature
-          property :author_signature
+          property :parent_author_signature, default: nil
+          property :author_signature, default: nil
+
+          # Generates XML and updates signatures
+          def to_xml
+            xml = entity_xml
+            hash = to_h
+            Relayable.update_signatures!(hash)
+
+            xml.at_xpath("author_signature").content = hash[:author_signature]
+            xml.at_xpath("parent_author_signature").content = hash[:parent_author_signature]
+            xml
+          end
```

Comment:
I'm not sure why you add this in the `class_eval`, there's nothing dynamic about it. Just add it as a regular instance method to the module.